gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/infra/adapter/PortalPageAdapter.java

```
@@ -0,0 +1,62 @@
+        return PortalPageFactory.createGraviteeMarkdownPage(entity.getId(), entity.getContent());
+    }
+
+    public static io.gravitee.repository.management.model.PortalPage toEntity(PortalPage domain, List<PortalViewContext> contexts) {
+        if (domain == null) return null;
+        io.gravitee.repository.management.model.PortalPage entity = new io.gravitee.repository.management.model.PortalPage();
+        entity.setId(domain.id().id().toString());
+        entity.setContent(domain.pageContent().content());
+        if (contexts != null) {
+            entity.setContexts(contexts.stream().map(Enum::name).collect(Collectors.toList()));
+        } else {
+            entity.setContexts(Collections.emptyList());
+        }
+        return entity;
+    }
+
+    public static List<PortalPage> toDomainList(List<io.gravitee.repository.management.model.PortalPage> entities) {
+        if (entities == null) return Collections.emptyList();
+        return entities.stream().map(PortalPageAdapter::toDomain).collect(Collectors.toList());
+    }
+
+    public static List<io.gravitee.repository.management.model.PortalPage> toEntityList(List<PortalPage> domains, List<List<PortalViewContext>> contextsList) {
+        if (domains == null) return Collections.emptyList();
+        return domains.stream().map(domain -> {
+            int idx = domains.indexOf(domain);
+            List<PortalViewContext> ctx = (contextsList != null && idx < contextsList.size()) ? contextsList.get(idx) : null;
+            return toEntity(domain, ctx);
+        }).collect(Collectors.toList());
+    }
+
+    public static List<PortalViewContext> toContextList(List<String> contextStrings) {
+        if (contextStrings == null) return Collections.emptyList();
+        return contextStrings.stream()
+            .map(s -> {
+                try {
+                    return PortalViewContext.valueOf(s);
+                } catch (Exception e) {
+                    return null;
```

WDYT of adding a log here for the silent error of a value mismatch?